### PR TITLE
St/shanghai fork sdk

### PR DESCRIPTION
**Pull request by saratnt**
## Description
Changes for Shanghai fork activation.

## Jira Ticket
[SDK-1594](https://horizenlabs.atlassian.net/browse/SDK-1594?atlOrigin=eyJpIjoiZDEzZjBhMTA2NmUyNGU1NGJmMzEwZjFjOTQwMTRiNTUiLCJwIjoiaiJ9)

## Changes
Added Version1_3 fork point and its mapping on go-ethereum Shanghai fork point.

## Checks
- [ X] Project Builds
- [ X] Project passes tests and checks
- [X ] Updated documentation accordingly
- [ X] Breaking changes have been correctly tagged and notified
 
## Additional information
It requires the following PRs:
- go-ethereum: https://github.com/HorizenOfficial/go-ethereum/pull/10
- libevm: https://github.com/HorizenOfficial/libevm/pull/26

[SDK-1594]: https://horizenlabs.atlassian.net/browse/SDK-1594?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: qa/SidechainTestFramework/scutil.py [hunk 1]
```diff
@@ -683,12 +683,18 @@ def start_sc_node(i, dirname, extra_args=None, rpchost=None, timewait=None, bina
...
+        if "-agentlib" in extra_args:
+            dbg_agent_opt = ' -agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005'
+        if "-mc_block_delay_ref" in extra_args:
+            additional_params = extra_args[extra_args.index("-mc_block_delay_ref") + 1]
+        if "-all_forks" in extra_args:
+            if additional_params != '':
+                additional_params = additional_params + " " + extra_args[extra_args.index("-all_forks") + 1]
+            else:
+                # Cmd line arguments are positional, so all_forks parameter needs to provide mc_block_delay_ref if not already set
+                additional_params = " 0 " + extra_args[extra_args.index("-all_forks") + 1]
```
Comment (i-skrypnyk) [inline qa/SidechainTestFramework/scutil.py:697]: :100:

**Review by i-skrypnyk — APPROVED** [review]

**Comment by i-skrypnyk** [comment]
I think the version 0.9.0 -> 0.10.0 should be also updated in `generate_report.sh` file

File: examples/account/evmapp/src/main/java/io/horizen/examples/EvmApp.java [hunk 2]
```diff
@@ -12,6 +12,9 @@ public class EvmApp {
     public static void main(String[] args) {
         if (args.length == 0) {
             System.out.println("Please provide settings file name as first parameter!");
+            System.out.println("Optional parameters:");
+            System.out.println("MC Block Reference delay (int)");
+            System.out.println("All forks enabled (boolean, default false");
```
Comment (paolocappelletti) [inline examples/account/evmapp/src/main/java/io/horizen/examples/EvmApp.java:17]: let's be a bit more clear:
All forks enabled from earliest epoch (2)
(also in the other config all the forks are enabled , but on different epochs..)

File: sdk/src/main/scala/io/horizen/account/fork/Version1_3_0Fork.scala [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package io.horizen.account.fork
+
+import io.horizen.fork.{ForkManager, OptionalSidechainFork}
+
+case class Version1_3_0Fork(active: Boolean = false) extends OptionalSidechainFork
+
+object Version1_3_0Fork {
```
Comment (paolocappelletti) [inline sdk/src/main/scala/io/horizen/account/fork/Version1_3_0Fork.scala:7]: please add a comment to explain what this fork enables
(see Version1_2_0Fork for reference)

**Review by paolocappelletti — CHANGES_REQUESTED** [review]

File: sdk/src/main/scala/io/horizen/account/fork/Version1_3_0Fork.scala (hunk 3 above)
Comment (saratnt) [reply sdk/src/main/scala/io/horizen/account/fork/Version1_3_0Fork.scala:13]: Done

File: examples/account/evmapp/src/main/java/io/horizen/examples/EvmApp.java (hunk 2 above)
Comment (saratnt) [reply examples/account/evmapp/src/main/java/io/horizen/examples/EvmApp.java]: Done

File: examples/account/evmapp/src/main/java/io/horizen/examples/AppForkConfiguratorAll.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
+package io.horizen.examples;
+
+import io.horizen.account.fork.*;
+import io.horizen.fork.*;
+import io.horizen.utils.Pair;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class AppForkConfiguratorAll extends ForkConfigurator {
```
[the remark this answers is not in the document]
Comment (saratnt) [reply examples/account/evmapp/src/main/java/io/horizen/examples/AppForkConfiguratorAll.java]: Done

**Review by paolocappelletti — APPROVED** [review]

**Pull request merged**